x-pack/plugins/apm/public/components/app/ServiceOverview/ServiceList/index.tsx

```diff
@@ -57,6 +58,27 @@ export const SERVICE_COLUMNS: Array<
       </EuiToolTip>
     )
   },
+  {
+    field: 'environments',
+    name: i18n.translate('xpack.apm.servicesTable.environmentColumnLabel', {
+      defaultMessage: 'Environment'
+    }),
+    width: '20%',
+    sortable: true,
+    render: (environments: string[]) => (
+      <TruncatedAnchorEuiToolTip
+        id="service-environments-tooltip"
+        content={environments.map(env => (
+          <React.Fragment key={env}>
+            {env}
+            <br />
+          </React.Fragment>
+        ))}
+      >
+        <>{environments.join(', ')}</>
+      </TruncatedAnchorEuiToolTip>
```

Comment:
What do you think about:
```tsx
<TruncatedEuiToolTip>
  <>{environments.join(', ')}</>
</TruncatedEuiToolTip>
```

**TruncatedEuiToolTip.tsx**
```tsx
function ToolTip({children, ...props}: { children: React.ReactElement} & Partial<EuiToolTipProps>) {
  return (
    <EuiToolTip content={children} {...props}>
      {children}
    </EuiToolTip>
  );
}

export const TruncatedEuiToolTip = styled(ToolTip)`
  ${truncate('100%')};
`;
```

This improves the types by just making it a light wrapper around `EuiToolTip` and simplifies how it is called.